moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1311,11 +1311,15 @@ as the new values that correspond to the group */
   /** \brief Update the state after setting a particular link to the input global transform pose.*/
   void updateStateWithLinkAt(const LinkModel* link, const Eigen::Isometry3d& transform, bool backward = false);
 
+  /** \brief Get the transform of the given link with respect to the root link of the RobotModel.
+      This is usually the same as the root link of the URDF unless a virtual joint was added. */
```

Comment:
```suggestion
```

Reply:
Well, I thought this was a helpful bit of advice. Will remove it if you insist.

Reply:
I like this comment because it's easy to check what the root link of the URDF is. The user can just print the tf tree, for example.

I think it's not so easy to check the root of the RobotModel. I see there's a method called getRootLink() but IMO the user shouldn't need to compile a program just to check the root link.

Reply:
Using "_usually_" the explanation is sloppy per definition, which should be avoided.
To become precise you should explain the "unusual" case as well. IMHO this will become too verbose for a function comment. Rather this higher-level concept should be explained e.g. in the tutorials.
The PlanningFrame is a well-defined concept for MoveIt.

Reply:
> IMHO this will become too verbose for a function comment.

I really like that Andy added this detail and I don't think we should nit-pick someone trying to document something that has been unclear for a long time. This is likely too detailed for a tutorial and the chances of making a tutorial on this are low.

@AndyZe sorry for making an overlapping PR https://github.com/ros-planning/moveit/pull/1846 this morning, I totally forgot this was the lack of documentation what you were complaining about yesterday! I will integrate your changes into mine now.

Reply:
I can follow the argument that the place to look for good practices should not be "random comments throughout the code base", but I am in favor of documenting and repeating common practices where they are relevant, rather than leaving them unsaid. It increases learnability unless it is misleading, which I don't think this is.